packages/compass-connections-navigation/src/connections-navigation-tree.tsx

```diff
@@ -0,0 +1,541 @@
+          setSize: collections.length,
+          posInSet: index + 1,
+        }))
+      : Array.from({ length: placeholdersLength }, (_, index) => ({
+          key: isInConnection
+            ? `${connectionIndex as number}-${databaseIndex}-${index}`
+            : `${databaseIndex}-${index}`,
+          level: level + 1,
+          type: 'placeholder' as const,
+        }))
+  );
+};
+
+const NavigationItem = memo<{
+  index: number;
+  style: React.CSSProperties;
+  data: ListItemData;
+}>(function NavigationItem({ index, style, data }) {
+  const {
+    items,
+    isLegacy,
+    isReadOnly,
+    activeNamespace,
+    currentTabbable,
+    onConnectionExpand,
+    onConnectionSelect,
+    onDatabaseExpand,
+    onNamespaceAction,
+  } = data;
+
+  const itemData = items[index];
+
+  if (itemData.type === 'connection') {
+    return (
+      <ConnectionItem
+        style={style}
+        isReadOnly={isReadOnly}
+        isLegacy={isLegacy}
+        isActive={activeNamespace === ''} // TODO(COMPASS-7775) we'll need something like activeConnection
+        isTabbable={itemData.id === currentTabbable}
+        onNamespaceAction={onNamespaceAction}
+        onConnectionExpand={onConnectionExpand}
+        onConnectionSelect={onConnectionSelect}
+        {...itemData}
+      ></ConnectionItem>
+    );
+  }
+
+  if (itemData.type === 'database') {
+    return (
+      <DatabaseItem
+        style={style}
+        isReadOnly={isReadOnly}
+        isLegacy={isLegacy}
+        isActive={itemData.id === activeNamespace}
+        isTabbable={itemData.id === currentTabbable}
+        onNamespaceAction={onNamespaceAction}
+        onDatabaseExpand={onDatabaseExpand}
+        {...itemData}
+      ></DatabaseItem>
+    );
+  }
+
+  return (
+    <div className={collectionItemContainer}>
+      <FadeInPlaceholder
+        isContentReady={itemData.type !== 'placeholder'}
+        contentContainerProps={{ style }}
+        fallbackContainerProps={{ style }}
+        content={() => {
+          return (
+            itemData.type !== 'placeholder' && (
+              <CollectionItem
+                isReadOnly={isReadOnly}
+                isLegacy={isLegacy}
+                isActive={itemData.id === activeNamespace}
+                isTabbable={itemData.id === currentTabbable}
+                onNamespaceAction={onNamespaceAction}
+                {...itemData}
+              ></CollectionItem>
+            )
+          );
+        }}
+        fallback={() => (
+          <PlaceholderItem
+            level={itemData.level}
+            isLegacy={isLegacy}
+          ></PlaceholderItem>
+        )}
+      ></FadeInPlaceholder>
+    </div>
+  );
+}, areEqual);
+
+const navigationTree = css({
+  flex: '1 0 auto',
+});
+
+const ConnectionsNavigationTree: React.FunctionComponent<{
+  connections?: Connection[];
+  databasesLegacy?: Database[];
+  expanded?: Record<string, false | Record<string, boolean>>;
+  expandedLegacy?: Record<string, boolean>;
```

Comment:
This feels like an unnecessary fork in logic here. We will need to support single connection in compass-web with the new sidebar: the logic is pretty straightforward, if there is only one connection and only one collection is allowed, this tree should render as before, can we just incorporate this logic into the component and not have a hard fork on this logic based on props? It's low effort and less cleaning-up afterwards when the multiple connections are enabled in Compass

Reply:
I agree, I'll do the change when I take this ticket.

Reply:
not sure if I follow here.. is the idea just to keep the `databases` name and `expanded` type will depend on whether `connections` or `databases` are set? we can do that for sure, it just doesn't seem like a "fork in logic" for me, it's really just the props layer. that's why I'm not sure if I understand correctly.
when I started on this I wasn't aware of the compass-web single connection scenario, so I thought the cleanup would be easiest if we clearly mark the legacy bits.

Reply:
also relevant note - the compass-web will likely keep using a separate sidebar because the new sidebar structure doesn't make sense for single connection. whether we call it legacy or single-connection (likely the latter, but I'm afraid to rename the sidebar folder at this moment, there are too many conflicts with the main branch already and at least one rebase last week apparently went wrong 🙈  )

Reply:
update: I've changed the props and also internally removed the legacy notation in favour of single connection (SC)